File: src/main/java/com/tyiu/corn/controller/TaskController.java
package com.tyiu.corn.controller;

import com.tyiu.corn.model.dto.TaskDTO;
import com.tyiu.corn.model.entities.Task;
import com.tyiu.corn.service.TaskService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

import java.util.List;

@RestController
@RequestMapping("/api/v1/task")
@RequiredArgsConstructor
public class TaskController {

    private final TaskService taskService;

    @GetMapping("/all")
    public Flux<TaskDTO> showListTask() {
        return taskService.getListTask();
    }

    @PostMapping("/add")
    public Mono<TaskDTO> addTask(@RequestBody TaskDTO task) {
        return taskService.saveTask(task);
    }

    @DeleteMapping("/delete/{id}")
    public Mono<Void> deleteTask(@PathVariable Long id) {
        taskService.deleteTask(id);
        return Mono.empty();
    }

    @PutMapping("/update/{id}")
    public Mono<Void> updateTask(@PathVariable Long id, @RequestBody TaskDTO updatedTask) {
        taskService.updateTask(id, updatedTask);
        return Mono.empty();
    }
}

File: scrum-service/src/main/kotlin/com/tyiu/scrumservice/controller/TaskController.kt
package com.tyiu.scrumservice.controller

import com.tyiu.scrumservice.model.*
import com.tyiu.scrumservice.service.TaskService
import kotlinx.coroutines.flow.Flow
import org.springframework.util.StopWatch.TaskInfo
import org.springframework.web.bind.annotation.*
import java.math.BigInteger

@RestController
@RequestMapping("/api/v1/scrum-service/task")
class TaskController (private val taskService: TaskService  ) {

    @GetMapping("/projects/{projectId}/all")
    fun getAllTaskByProject(@PathVariable projectId: String): Flow<TaskDTO> = taskService.getAllTasksByProject(projectId)

    @GetMapping("/projects/{projectId}")
    fun getAllTasksInBackLog(@PathVariable projectId: String): Flow<TaskDTO> = taskService.getAllTasksInBacklog(projectId)

    @GetMapping("/projects/{projectId}/sprint/{sprintId}")
    fun getAllTasksInSprint(@PathVariable projectId: String, @PathVariable sprintId: String): Flow<TaskDTO> = taskService.getAllTasksInSprint(projectId, sprintId)

    @GetMapping("/{id}")
    fun getOneTaskById(@PathVariable id: String): Flow<TaskDTO> = taskService.getOneTaskById(id)

    @PostMapping("/add")
    suspend fun postCreateTask(@RequestBody taskCreateRequest: TaskCreateRequest): Unit = taskService.createTask(taskCreateRequest)

    @PutMapping("/status/change")
    suspend fun putTaskStatus (@RequestBody taskStatusRequest: TaskStatusRequest) = taskService.putTaskStatus(taskStatusRequest)

    @PutMapping("/{taskId}/update")
    suspend fun putUpdateTask (@PathVariable taskId: String,@RequestBody taskInfoRequest: TaskInfoRequest) = taskService.putUpdateTask(taskId, taskInfoRequest)

    @DeleteMapping("/{id}/delete")
    suspend fun deleteTask(@PathVariable id: String) = taskService.deleteTask(id)
}